### DISCUSSION TITLE
✅ Site-Wide Reusable Section/Container Blocks for Elementor

### DISCUSSION THREAD

**Discussion by craiggarrity** [elementor/elementor#24233] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

The primary problem this feature request aims to solve is the lack of flexibility and efficiency in managing and updating reusable content (specifically sections/containers) across multiple pages on a website built with Elementor. As far as I know, we can only save global widgets (which are the element) not the parent section/container settings. I'm aware we can save sections as templates, but again if you want to add them in on pages, its a manual method and it doesn't update all instances if any changes were to be made in future.

Currently, if a user needs to add a specific content section (e.g., a call-to-action button, a promotional banner, a floating WhatsApp icon) to multiple pages, they have to manually edit each page to include this section. This process is time-consuming and impractical, especially for websites with a large number of pages.

Furthermore, if there's a need to update this content later (for instance, changing the content), the user must again manually make this change on every individual page where the content appears. This process is not only inefficient but also prone to inconsistencies and errors.

In summary, the problem is the absence of a feature that allows for the creation, management, and updating of standalone reusable content blocks that can be applied and updated site-wide from a single location, similar to what is available for global headers and footers in Elementor.

### Describe the solution you'd like

I would like to create a modular/reusable section with containers and content that would enable users to create a modular piece of content (e.g., a custom-designed container, a floating contact button, a promotional banner) and set conditions for it to appear across selected or all pages. This feature should ideally allow for:

- Creating a standalone, customizable content block.
- Setting display conditions (e.g., all pages, specific pages, excluding certain pages).
- Easy updating of the content block from a single location, automatically reflecting changes site-wide.
- Integration with existing Elementor widgets for versatile content creation.

### Describe alternatives you've considered

Other website builders, like Astra with its custom layouts, already offer this kind of functionality. Extending Elementor's existing Theme Builder and global widget capabilities to include this feature would significantly enhance its competitiveness and utility.

**Comment by 79ho3ein** [comment]
https://github.com/orgs/elementor/discussions/23622

**Comment by saawood13** [comment]
> Easy updating of the content block from a single location, automatically reflecting changes site-wide.

This!! I have been trying to figure out how to do this, assumed it would have already been a feature

**Comment by TorqueMedia** [comment]
Just came here to request this too! Ohh the efficiency I could use right now across 54 pages.

**Comment by nicholaszein** [comment]
Hello everyone!

This feature already exists with both [Elementor Shortcodes](https://elementor.com/help/how-to-use-shortcodes/#elementor-shortcodes), and the [Template widget](https://elementor.com/help/template-widget-pro/) (Pro).

With these two options you can show any Saved Template from the Template Library on any page. This means, they behave exactly the same as a Header, Footer, or any other Theme Builder templates. You edit them directly from the library, and they update on every page. This is the same behavior of the Global Widgets by the way, just adapted to be used for Library Templates.

Instead of importing a template from the library to your posts and pages, you simply use the Template widget (or a template shortcode), and can update it in one place to be applied to all instances added to your different documents on your website.

You should really take a look, this is a cool feature!

I'm closing this request as solved.

Kind regards
